tests/cli/test_utils.py

```diff
@@ -101,3 +113,190 @@ def test_validate_with_invalid_directory_if_default_is_valid(tmp_path: pathlib.P
+    )
+
+    config_data = rasa.shared.utils.io.read_yaml_file(config_path)
+
+    for k in parameters["mandatory_keys"]:
+        assert k in config_data
+
+
+@pytest.mark.parametrize(
+    "parameters",
+    [
+        {
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS,
+        },
+        {
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_CORE,
+        },
+    ],
+)
+def test_get_validated_config_with_default_config(parameters: Dict[Text, Any]) -> None:
+    config_path = None
+
+    default_config_path = os.path.join(tempfile.mkdtemp(), "default-config.yml")
+    rasa.shared.utils.io.write_yaml(parameters["default_config"], default_config_path)
+
+    config_path = rasa.cli.utils.get_validated_config(
+        config_path, parameters["mandatory_keys"], default_config_path
+    )
+
+    config_data = rasa.shared.utils.io.read_yaml_file(config_path)
+
+    for k in parameters["mandatory_keys"]:
+        assert k in config_data
+
+
+@pytest.mark.parametrize(
+    "parameters",
+    [
+        {
+            "config_data": {
+                "assistant_id": "placeholder_default",
+            },
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS,
+        },
+        {
+            "config_data": {
+                "assistant_id": "placeholder_default",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+                "imports": "other-folder",
+            },
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "language": "en",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_NLU,
+        },
+        {
+            "config_data": None,
+            "default_config": {
+                "assistant_id": "placeholder_default",
+                "pipeline": "supervised",
+                "policies": ["TEDPolicy", "FallbackPolicy"],
+            },
+            "mandatory_keys": CONFIG_MANDATORY_KEYS_NLU,
+        },
+        {
+            "config_data": None,
+            "default_config": None,
+            "mandatory_keys": CONFIG_MANDATORY_KEYS,
+        },
+    ],
+)
+def test_get_validated_config_with_invalid_input(parameters: Dict[Text, Any]) -> None:
```

Comment:
We should split this test into several tests to avoid if-else branching in tests.
We can do it in a new PR after this one is merged.

Reply:
Should this be a new subtask for the parent user story?